### PR TITLE
BUG: Indexing a timestamp ArrowDtype Index

### LINKED ISSUE #53644: BUG: [pyarrow] AttributeError: 'Index' object has no attribute 'freq'
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import numpy as np
import pandas as pd


t = pd.date_range("2023-01-01", "2023-01-04", freq="1h")
x = np.random.randn(len(t))
df = pd.DataFrame(x, index=t, columns=["values"])

df.to_parquet("frame.parquet")
df2 = pd.read_parquet("frame.parquet", dtype_backend="pyarrow")

df.loc[df.index[:-5]]  # ✔
df2.loc[df2.index[:-5]]  # ✘ AttributeError: 'Index' object has no attribute 'freq'
```

### Issue Description

Indexing an index of type `timestamp[us][pyarrow]` with an index of type `timestamp[us][pyarrow]` fails with `AttributeError: 'Index' object has no attribute 'freq'`.

The issue seems to be occurring here:

https://github.com/pandas-dev/pandas/blob/4149f323882a55a2fe57415ccc190ea2aaba869b/pandas/core/indexes/base.py#L6039-L6046

### Expected Behavior

It should yield the same result independent of the dtype backend.

### Installed Versions

<details>

```
INSTALLED VERSIONS
------------------
commit           : 965ceca9fd796940050d6fc817707bba1c4f9bff
python           : 3.11.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.19.0-43-generic
Version          : #44~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon May 22 13:39:36 UTC 2
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.2
numpy            : 1.23.5
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.7.2
pip              : 23.1.2
Cython           : 0.29.34
pytest           : 7.3.1
hypothesis       : None
sphinx           : 7.0.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.2
html5lib         : None
pymysql          : 1.0.3
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.13.2
pandas_datareader: None
bs4              : 4.12.2
bottleneck       : None
brotli           : None
fastparquet      : 2023.4.0
fsspec           : 2023.5.0
gcsfs            : None
matplotlib       : 3.7.1
numba            : 0.57.0
numexpr          : 2.8.4
odfpy            : None
openpyxl         : 3.1.2
pandas_gbq       : None
pyarrow          : 12.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.10.1
snappy           : None
sqlalchemy       : 1.4.48
tables           : 3.8.0
tabulate         : 0.9.0
xarray           : 2023.4.2
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None
```

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #53644 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -6037,7 +6038,7 @@ def _get_indexer_strict(self, key, axis_name: str_t) -> tuple[Index, np.ndarray]
         if isinstance(key, Index):
             # GH 42790 - Preserve name from an Index
             keyarr.name = key.name
-        if keyarr.dtype.kind in "mM":
+        if keyarr.dtype.kind in "mM" and not isinstance(keyarr.dtype, ArrowDtype):
```
Comment (jbrockmendel) [inline pandas/core/indexes/base.py:6041]: i think use `lib.is_np_dtype(keyarr.dtype, "mM")`?

Comment (mroeschke) [reply pandas/core/indexes/base.py:6041]: I would have to include `DatetimeTZDtype` here then too based on failing tests so is `lib.is_np_dtype(keyarr.dtype, "mM") or isinstance(keyarr.dtype, DatetimeTZDtype)` better?

Comment (jbrockmendel) [reply pandas/core/indexes/base.py:6041]: good point.  yes, i think that would be more explicit about the cases that are supposed to go through here

File: pandas/tests/indexing/test_datetime.py [hunk 2]
```diff
@@ -168,3 +168,21 @@ def test_getitem_str_slice_millisecond_resolution(self, frame_or_series):
             ],
         )
         tm.assert_equal(result, expected)
+
+    def test_getitem_pyarrow_index(self, frame_or_series):
+        # GH 53644
+        pytest.importorskip("pyarrow")
```
Comment (jbrockmendel) [inline pandas/tests/indexing/test_datetime.py:174]: isnt there a decorator for this?

**Comment by jbrockmendel** [comment]
small comment, also needs rebase.  otherwise LGTM

File: pandas/tests/indexing/test_datetime.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/indexing/test_datetime.py:174]: I think there is, but I think that decorator can be eventually replaced with this builtin pytest functionality

**Pull request merged**